lumigator/backend/backend/tests/integration/api/routes/test_api_workflows.py

```diff
@@ -214,75 +206,137 @@ def test_upload_data_no_gt_launch_annotation(
 def test_full_experiment_launch(
     local_client: TestClient, dialog_dataset, dependency_overrides_services
 ):
+    """This is the main integration test: it checks:
```

Comment:
Maybe we could put each of these items also as comments at the start of each section? Make separate functions?

Reply:
Ooh this is a good idea. I'll try to make the comments more clean.

Reply:
TBH this is much better than my default level of commenting ^_^